source/simulator/assembly.cc

```diff
@@ -756,6 +772,12 @@ namespace aspect
         scratch.finite_element_values[solution_field].get_function_laplacians (old_old_solution,
                                                                                scratch.old_old_field_laplacians);
 
+        if (parameters.include_melt_transport && melt_handler->is_porosity(advection_field))
+          {
+            scratch.finite_element_values[introspection.extractors.velocities].get_function_divergences (current_linearization_point,
+                scratch.current_velocity_divergences);
+          }
```

Comment:
Hm, this is of course unpleasant. But I don't really see how to avoid it either.